polaris-react/src/tokens/token-groups/opacity.json

```diff
@@ -0,0 +1,23 @@
+{
+  "opacity-0": "0",
+  "opacity-1": "0.05",
+  "opacity-2": "0.1",
+  "opacity-3": "0.15",
+  "opacity-4": "0.2",
+  "opacity-5": "0.25",
+  "opacity-6": "0.3",
+  "opacity-7": "0.35",
+  "opacity-8": "0.4",
+  "opacity-9": "0.45",
+  "opacity-10": "0.5",
+  "opacity-11": "0.55",
+  "opacity-12": "0.6",
+  "opacity-13": "0.65",
+  "opacity-14": "0.7",
+  "opacity-15": "0.75",
+  "opacity-16": "0.8",
+  "opacity-17": "0.85",
+  "opacity-18": "0.9",
+  "opacity-19": "0.95",
+  "opacity-20": "1"
```

Comment:
I think 20 values is too many. I'm not sure it adds much value to tokenize them. Essentially we are just saying "stick to one decimal place or halves" and from doing a quick search in `web`, that seems to already be an established pattern.

Reply:
What would you recommend the amount of values be @aveline?

Reply:
I am actually questioning the value of having opacity tokens at all. So often the "right" opacity is so contextual based on the colours used and background.

I could see valid use cases for all the values @aaronccasanova is including above, but just less convinced that mapping them to tokens adds value.

Tokens can help narrow down possibilities of values, for example a collection of colour tokens narrows it down from 16+ million options. The options for opacity are already pretty limited in that it needs to be a value between `1` and `0`. Do we need to encourage consumers to not set an opacity of `0.233645` when it doesn't seem like they are trying to do that anyway? 🤷‍♀️

If anything I see fewer than 20 opacity values in the web codebase right now. So why create tokens for values no one may ever use?

Although maybe we do want to prevent the few random `opacity: 100` 😬

Reply:
From a quick pass through web I ended up finding 19 different (17 not including 1 and 0) opacity values: 

- 0
- 0.02
- 0.10
- 0.20
- 0.25
- 0.30
- 0.35
- 0.40
- 0.43
- 0.45
- 0.50
- 0.60
- 0.70
- 0.80
- 0.85
- 0.90
- 0.95
- 1
- 100

Even though it's a lot of values I do think it helps to keep some consistency and give guidance/an opinion on how we want to handle opacity. There were a few instances in Polaris in this PR and web that are more random / don't follow the 0.05 step convention.

Reply:
Do we expect to ship a change where we would make the UI more opaque or transparent. If the answer is no then I would consider letting opacity be an ignored value in coverage and not adding tokens. I feel it is too early to know if we get value from controlling opacity from the design system.